File: ios/StockPro/Data/DTOs/InvoiceDTOs.swift
import Foundation

struct InvoiceDTO: Decodable, Identifiable {
    let id: Int
    let invoice_number: String?
    let customer_id: Int?
    let customer_name: String?
    let status: String
    let total: Double
    let payment_method: String?
    let notes: String?
    let due_date: String?
    let created_at: String
    let paid_at: String?

    var statusLabel: String {
        switch status {
        case "ticket": "Ticket de caisse"
        case "brouillon": "Brouillon"
        case "envoyee": "Envoyée"
        case "payee": "Payée"
        case "annulee": "Annulée"
        default: status
        }
    }
}

struct InvoiceItemDTO: Decodable, Identifiable {
    let id: Int
    let product_id: Int
    let product_name: String?
    let quantity: Int
    let price: Double
    let total: Double?

    enum CodingKeys: String, CodingKey {
        case id, product_id, product_name, quantity
        case price = "unit_price"
        case total = "line_total"
    }
}

struct InvoiceCreateRequest: Encodable {
    let customer_id: Int?
    let items: [InvoiceItemRequest]
    let payment_method: String?
    let notes: String?
}

struct InvoiceItemRequest: Encodable {
    let product_id: Int
    let quantity: Int
    let price: Double
}

struct InvoiceUpdateRequest: Encodable {
    let status: String?
    let notes: String?
}

File: App2/core/data/src/main/java/com/app2/core/data/remote/dto/InvoiceDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class InvoiceDTO(
    val id: Int,
    @SerialName("invoice_number") val invoiceNumber: String? = null,
    @SerialName("customer_id") val customerId: Int? = null,
    @SerialName("warehouse_id") val warehouseId: Int? = null,
    val status: String? = "brouillon",
    val subtotal: Double? = null,
    @SerialName("discount_total") val discountTotal: Double? = null,
    @SerialName("tax_amount") val taxAmount: Double? = null,
    val total: Double? = null,
    val notes: String? = null,
    @SerialName("due_date") val dueDate: String? = null,
    @SerialName("paid_at") val paidAt: String? = null,
    @SerialName("customer_name") val customerName: String? = null,
    @SerialName("payment_method") val paymentMethod: String? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class InvoiceItemDTO(
    val id: Int,
    @SerialName("invoice_id") val invoiceId: Int? = null,
    @SerialName("product_id") val productId: Int? = null,
    @SerialName("product_name") val productName: String? = null,
    @SerialName("product_sku") val productSku: String? = null,
    val quantity: Int,
    @SerialName("unit_price") val unitPrice: Double,
    @SerialName("discount_percent") val discountPercent: Double? = null,
    @SerialName("tax_rate") val taxRate: Double? = 20.0,
    @SerialName("line_total") val lineTotal: Double,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class InvoiceCreateRequest(
    @SerialName("customer_id") val customerId: Int? = null,
    val status: String? = null,
    val notes: String? = null,
    @SerialName("due_date") val dueDate: String? = null
)

@Serializable
data class InvoiceItemRequest(
    @SerialName("product_id") val productId: Int,
    val quantity: Int,
    @SerialName("unit_price") val unitPrice: Double,
    @SerialName("discount_percent") val discountPercent: Double? = null
)
